teps/0094-configuring-resources-at-runtime.md

```diff
@@ -52,12 +61,476 @@ While this may be a valuable feature, it should be considered in a separate prop
 depending on the image or source being built.
 - Kubeflow pipelines and other data pipelines may have variable resource requirements
 depending on the data being processed.
-- Catalog Tasks should be generally reusable in different environments
+- Catalog `Task`s should be generally reusable in different environments
 that may have different resource constraints.
 
 ## Requirements
 
 - Users can specify `Step` and `Sidecar` resource requirements at runtime.
 - Users can specify `Step` and `Sidecar` resource requirements for `Task`s 
 or `Pipeline`s they don't own, especially those in the Catalog.
-- Users can specify resource requirements for individual `Step`s and `Sidecar`s.
\ No newline at end of file
+- Users can specify resource requirements for individual `Step`s and `Sidecar`s.
+
+## Proposal
+
+Augment `TaskRunSpec` and `PipelineTaskRunSpec` with a mapping of `Step` names to overrides
+and a mapping of `Sidecar` names to overrides.
+
+## Design Details
+
+```go
+import corev1 "k8s.io/api/core/v1"
+
+type TaskRunStepOverride struct {
+  // The name of the Step to override.
+  TaskRunStepName string
```

Comment:
```suggestion
  Name string
```

general UX nit: `TaskRunStepName` nested within `TaskRunStepOverrides` is a bit repetitive/redundant. We could probably slim this down to just `Name` without confusion about what this refers to. Godoc also helps clearly define what this should reference.


e.g.

```yaml
apiVersion: tekton.dev/v1beta1
kind: TaskRun
metadata:
  name: image-build-taskrun
spec:
  taskRef:
    name: image-build-task
  stepOverrides:
    - name: build
      resources:
        requests:
          memory: 1Gi
```

This is line with other types like [TaskRef](https://pkg.go.dev/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1#TaskRef)/[PipelineRef](https://pkg.go.dev/github.com/tektoncd/pipeline/pkg/apis/pipeline/v1beta1#PipelineRef) (i.e. they don't use `TaskName`)

Reply:
done